std/manual.md

```diff
@@ -1210,6 +1210,61 @@ console.log(result["/foo.ts"].map);
 We would expect the `enum` would be rewritten to an IIFE which constructs the
 enumerable, and the map to be defined.
 
+## TypeScript Compiler Options
+
+In Deno ecosystem, all strict flags are enabled in order to comply with TypeScript
+ideal of being `strict` by default. However, in order to provide a way to support customization a configuration file such as `tsconfig.json` might be provided to Deno on
+program execution.
+
+You do need to explicitly tell Deno where to look for this configuration, in order to do so
+you can use the `-c` argument when executing your application.
+
+```bash
+deno -c tsconfig.json mod.ts
+```
+
+Currently allowed settings, as well as their default values in Deno go as follows:
+
+```json
+{
+   "compilerOptions": {
+      "allowJs": false,
+      "allowUmdGlobalAccess": false,
+      "allowUnreachableCode": false,
+      "allowUnusedLabels": false,
+      "alwaysStrict": true,
+      "assumeChangesOnlyAffectDirectDependencies": false,
```

Comment:
Hmmm... there are some new compiler options here that I need to have a look at, like this one.  I am not sure we have the right settings, and they should be hard-coded and not left up to manipulation.

Reply:
Yep. In our Gitter discussion earlier you mentioned the non-negotiable options Deno ignores to avoid breaking things.

So I removed those options and compared with Typescript docs to check the default values of the remaining ones.

In order to check the possible impact of the new flags, we could set them to the alternative of the default and run the tests(chances are its way more complex than this and we end up adding test cases for these edge cases)

Reply:
I think having them in here is great now, because if we "hard code" them in the future, we can reflect the changes here as well.  Thanks for the effort.

Reply:
All ok @kitsonk! Added "**lib**" as an options since #4234 was resolved.